pkg/network/network_identity.go

```diff
@@ -0,0 +1,209 @@
+
+		webhookDeployment := &appsv1.Deployment{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Deployment",
+				APIVersion: appsv1.SchemeGroupVersion.String(),
+			},
+		}
+		nsn := types.NamespacedName{Namespace: hcpCfg.Namespace, Name: "network-identity"}
+		if err := client.Default().CRClient().Get(context.TODO(), nsn, webhookDeployment); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return nil, fmt.Errorf("failed to retrieve existing network-identity deployment: %w", err)
+			} else {
+				klog.Infof("network-identity deployment does not exist")
+			}
+		} else {
+			webhookReady = !deploymentProgressing(webhookDeployment)
+		}
+
+		manifestDirs = append(manifestDirs, filepath.Join(manifestDir, "network-identity/managed"))
+	} else {
+		// self-hosted specific
+		data.Data["NetworkIdentityImage"] = os.Getenv("OVN_IMAGE")
+
+		// NetworkIdentityTerminationDurationSeconds holds the allowed termination duration
+		// During node reboot, the webhook has to wait for the API server to terminate first to avoid disruptions
+		data.Data["NetworkIdentityTerminationDurationSeconds"] = 200
+
+		apiServer := bootstrapResult.Infra.APIServers[bootstrap.APIServerDefault]
+		data.Data["K8S_APISERVER"] = "https://" + net.JoinHostPort(apiServer.Host, apiServer.Port)
+
+		// NetworkIdentityAddress is only used in self-hosted deployments where the webhook listens on loopback
+		// listening on localhost always picks the v4 address while dialing to localhost can choose either one
+		// https://github.com/golang/go/issues/9334
+		// For that reason set the webhook address use the loopback address of the primary IP family
+		// Note: ServiceNetwork cannot be empty, so it is safe to use the first element
+		data.Data["NetworkIdentityAddress"] = "127.0.0.1"
+		if utilnet.IsIPv6CIDRString(conf.ServiceNetwork[0]) {
+			data.Data["NetworkIdentityAddress"] = "[::1]"
+		}
+
+		var err error
+		clusterBootstrapFinished, err = isBootstrapComplete(client)
+		if err != nil {
+			return nil, err
+		}
+
+		webhookDaemonSet := &appsv1.DaemonSet{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "DaemonSet",
+				APIVersion: appsv1.SchemeGroupVersion.String(),
+			},
+		}
+		nsn := types.NamespacedName{Namespace: NetworkIdentityNamespace, Name: "network-identity"}
+		if err := client.Default().CRClient().Get(context.TODO(), nsn, webhookDaemonSet); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return nil, fmt.Errorf("failed to retrieve existing network-identity daemonset: %w", err)
+			} else {
+				klog.Infof("network-identity daemonset does not exist")
+			}
+		} else {
+			webhookReady = !daemonSetProgressing(webhookDaemonSet, false)
+		}
+
+		manifestDirs = append(manifestDirs, filepath.Join(manifestDir, "network-identity/self-hosted"))
+	}
+
+	var webhookCA []byte
+	if err := webhookCAClient.CRClient().Get(context.TODO(), webhookCALookup, webhookCAConfigMap); err != nil {
+		if !apierrors.IsNotFound(err) {
```

Comment:
not clear on what happens if this is not found, we end up with an empty CA bundle and we don't care about that?

Reply:
The webhook won't be rendered in that case, I will add a comment.